aws/logs_monitoring/lambda_function.py

```diff
@@ -122,11 +122,34 @@ def __init__(self, name, pattern, placeholder):
             "(?<!^)\s+(?={})(?!.\s)".format(DD_MULTILINE_LOG_REGEX_PATTERN)
         )
     except Exception:
-        raise Exception("could not compile multiline regex with pattern: {}".format(DD_MULTILINE_LOG_REGEX_PATTERN))
+        raise Exception("could not compile s3 multiline regex with pattern: {}".format(DD_MULTILINE_LOG_REGEX_PATTERN))
     multiline_regex_start_pattern = re.compile(
         "^{}".format(DD_MULTILINE_LOG_REGEX_PATTERN)
     )
 
+
+# DD_MULTILINE_CLOUDWATCH_REGEX: Datadog Multiline Cloudwatch Logs Regular Expression Patterns for Log Groups
+DD_MULTILINE_CLOUDWATCH_LOG_REGEX_PATTERNS = None
+if "DD_MULTILINE_CLOUDWATCH_LOG_REGEX_PATTERNS" in os.environ:
+    multiline_regexs = os.environ["DD_MULTILINE_CLOUDWATCH_LOG_REGEX_PATTERNS"]
+    try:
+        DD_MULTILINE_CLOUDWATCH_LOG_REGEX_PATTERNS = {}
+        cloudwatch_regex_dict = json.loads(multiline_regexs)
+
+        # compile regex start paterns for each dictionary entry
+        for log_group_name in cloudwatch_regex_dict:
+            if log_group_name not in DD_MULTILINE_CLOUDWATCH_LOG_REGEX_PATTERNS:
+                log_group_rule = cloudwatch_regex_dict[log_group_name]
+
+                # ensure pattern only matchs at start of line
+                DD_MULTILINE_CLOUDWATCH_LOG_REGEX_PATTERNS[log_group_name] = re.compile(
+                    "^{}".format(log_group_rule)
```

Comment:
this code is duplicated from https://github.com/DataDog/datadog-serverless-functions/pull/129/files#diff-618731ddf1c446ef45d09613cb4b189bR127, do we want to factorise ?

Reply:
👍 makes sense, abstracted this to it's own function [`create_multiline_start_regex`](https://github.com/DataDog/datadog-serverless-functions/blob/c1f2d4dceae006868c6b9697fe7b5919b0da83c6/aws/logs_monitoring/lambda_function.py#L117)